app.py

```diff
@@ -1,57 +1,159 @@
-        # x = Office('Test Office')
-        # y = LivingSpace('Test Living Spaces')
-        # a = Fellow('Ahmed')
-        # b = Staff('Yusuf')
-        # a.accom = 'Y'
-        # c = Fellow('Sebit')
-        # d = Staff('Rajab')
-        # g = Amity()
-        #
-        #
-        # x.occupants.append(a)
-        # y.occupants.append(b)
-        # x.occupants.append(c)
-        # y.occupants.append(d)
-        #
-        # g.all_rooms.append(x)
-        # g.all_rooms.append(y)
-        #
-        # print (x.room_name, x.room_type, len(x.occupants))
-        # print (y.room_name, y.room_type, len(y.occupants))
-        # print (a.person_name, a.person_type, a.accom)
-        # print (len(g.all_rooms))
-        #
-        # h = [x.room_name for x in g.all_rooms]
-        #
-        # if 'Test Offices' in h:
-        #     print ('Exists')
-        # else:
-        #     print ('Doesnt exists')
-
-        amity_controls = AmityControls()
-        amity = Amity()
-        amity_controls.create_room('Amity', 'Office')
-        amity_controls.create_room('AmityLiving', 'Living Space')
-        amity_controls.create_room('AmityOff', 'Office')
-        amity_controls.create_room('AmityLiv', 'Living Space')
-        amity_controls.add_person('Ahmed', 'Yusuf', 'Fellow', 'Y')
-        offices = [x for x in amity_controls.rooms if x.room_type == 'OFFICE']
-        livingspaces = [x for x in amity_controls.rooms if x.room_type == "LIVING SPACE"]
-        print (amity_controls.persons)
-        print (amity_controls.rooms)
-        print (offices)
-        print (livingspaces)
-amitymain = AmityMain()
-amitymain.person()
+
+def docopt_cmd(func):
+
+    """
+    This decorator is used to simplify the try/except block and pass the result
+    of the docopt parsing to the called action.
+    """
+    def fn(self, arg):
+        try:
+            opt = docopt(fn.__doc__, arg)
+        except DocoptExit as e:
+            # The DocoptExit is thrown when the args do not match.
+            # We print a message to the user and the usage block.
+            print('Invalid Command!')
+            print(e)
+            return
+        except SystemExit:
+            # The SystemExit exception prints the usage for --help
+            # We do not need to do the print here.
+            return
+        return func(self, opt)
+    fn.__name__ = func.__name__
+    fn.__doc__ = func.__doc__
+    fn.__dict__.update(func.__dict__)
+    return fn
+
+
+
+class MyInteractive(cmd.Cmd):
+    amity_controls = AmityControls()
+    cprint(figlet_format("Amity", font="isometric4"), attrs=['bold'])
+    prompt = '(app) '
+    file = None
+    print(__doc__)
+
+
+    @docopt_cmd
+    def do_create_room(self, args):
+
+        """Usage: create_room <room_type> <room_name>..."""
+        room_type = args['<room_type>'].upper()
+        for name in args['<room_name>']:
+            room_name = args['<room_name>'][args['<room_name>'].index(name)]
```

Comment:
I don't quite understand why you are getting the index of `room_name`. Kindly explain